dna/zomes/courses/code/src/course.rs

```diff
@@ -259,15 +436,87 @@ pub fn add_module_to_course(
     course_address: &Address,
     module_address: &Address,
 ) -> ZomeApiResult<Address> {
-    let current_course = hdk::get_entry(course_address).unwrap().unwrap();
-    if let Entry::App(_, current_course) = current_course {
-        let mut course_entry = Course::try_from(current_course.clone())
+    let current_course_anchor = hdk::get_entry(course_address).unwrap().unwrap();
+
+    // QUESTION: why are we doing it like that?
```

Comment:
and this one is more about code consistency. We're treating entry retrieval differently in `update` and `get_my_courses` methods and I just want to understand which method is more appropriate. 
@HedayatAbedijoo @guillemcordoba could you please clarify that?

Reply:
UPD: it has been clarified by @HedayatAbedijoo that since this is an educational app, we're just showing different ways of how something can be done.

Reply:
UPD: it has been clarified by @HedayatAbedijoo that since this is an educational app, we're just showing different ways of how something can be done. So it's ok that it looks inconsistent.

Reply:
I still think it is good to have both version with comment above to show all options. but feel free to follow your taste.